From c0ceaa6d6231cc567e820fd34e27c641870daa32 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 29 Aug 2010 15:40:10 +0100
Subject: [PATCH] r8192u_usb: Don't export symbols from ieee80211 stack

This is a private (and modified) copy of the old ieee80211 stack which
should not be exported to other modules.
---
 drivers/staging/rtl8192u/ieee80211/api.c           |    2 ++
 drivers/staging/rtl8192u/ieee80211/dot11d.c        |    2 ++
 drivers/staging/rtl8192u/ieee80211/ieee80211_rx.c  |    2 ++
 .../staging/rtl8192u/ieee80211/ieee80211_softmac.c |    2 ++
 .../rtl8192u/ieee80211/ieee80211_softmac_wx.c      |    2 ++
 drivers/staging/rtl8192u/ieee80211/ieee80211_tx.c  |    2 ++
 drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c  |    6 ++++++
 .../staging/rtl8192u/ieee80211/rtl819x_HTProc.c    |    2 ++
 .../staging/rtl8192u/ieee80211/rtl819x_TSProc.c    |    2 ++
 9 files changed, 22 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/rtl8192u/ieee80211/api.c b/drivers/staging/rtl8192u/ieee80211/api.c
index c627d02..ecf515a 100644
--- a/drivers/staging/rtl8192u/ieee80211/api.c
+++ b/drivers/staging/rtl8192u/ieee80211/api.c
@@ -239,8 +239,10 @@ EXPORT_SYMBOL_GPL(crypto_free_tfm);
 EXPORT_SYMBOL_GPL(crypto_alg_available);
 */
 
+/*
 EXPORT_SYMBOL_NOVERS(crypto_register_alg);
 EXPORT_SYMBOL_NOVERS(crypto_unregister_alg);
 EXPORT_SYMBOL_NOVERS(crypto_alloc_tfm);
 EXPORT_SYMBOL_NOVERS(crypto_free_tfm);
 EXPORT_SYMBOL_NOVERS(crypto_alg_available);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/dot11d.c b/drivers/staging/rtl8192u/ieee80211/dot11d.c
index b91cbfc..6444dab 100644
--- a/drivers/staging/rtl8192u/ieee80211/dot11d.c
+++ b/drivers/staging/rtl8192u/ieee80211/dot11d.c
@@ -212,6 +212,7 @@ int ToLegalChannel(
 
 	return default_chn;
 }
+/*
 EXPORT_SYMBOL(Dot11d_Init);
 EXPORT_SYMBOL(Dot11d_Reset);
 EXPORT_SYMBOL(Dot11d_UpdateCountryIe);
@@ -219,5 +220,6 @@ EXPORT_SYMBOL(DOT11D_GetMaxTxPwrInDbm);
 EXPORT_SYMBOL(DOT11D_ScanComplete);
 EXPORT_SYMBOL(IsLegalChannel);
 EXPORT_SYMBOL(ToLegalChannel);
+*/
 
 #endif
diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_rx.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_rx.c
index 0e003c5..f2f133e 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_rx.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_rx.c
@@ -2664,5 +2664,7 @@ void ieee80211_rx_mgt(struct ieee80211_device *ieee,
 	}
 }
 
+/*
 EXPORT_SYMBOL(ieee80211_rx_mgt);
 EXPORT_SYMBOL(ieee80211_rx);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c
index 9cce5bd..fb3bc1a 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac.c
@@ -3266,6 +3266,7 @@ void notify_wx_assoc_event(struct ieee80211_device *ieee)
 	wireless_send_event(ieee->dev, SIOCGIWAP, &wrqu, NULL);
 }
 
+/*
 EXPORT_SYMBOL(ieee80211_get_beacon);
 EXPORT_SYMBOL(ieee80211_wake_queue);
 EXPORT_SYMBOL(ieee80211_stop_queue);
@@ -3286,4 +3287,5 @@ EXPORT_SYMBOL(ieee80211_stop_scan);
 EXPORT_SYMBOL(ieee80211_send_probe_requests);
 EXPORT_SYMBOL(ieee80211_softmac_scan_syncro);
 EXPORT_SYMBOL(ieee80211_start_scan_syncro);
+*/
 //EXPORT_SYMBOL(ieee80211_sta_ps_send_null_frame);
diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac_wx.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac_wx.c
index f335c25..028d949 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac_wx.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_softmac_wx.c
@@ -595,6 +595,7 @@ exit:
 	return ret;
 
 }
+/*
 EXPORT_SYMBOL(ieee80211_wx_get_essid);
 EXPORT_SYMBOL(ieee80211_wx_set_essid);
 EXPORT_SYMBOL(ieee80211_wx_set_rate);
@@ -613,3 +614,4 @@ EXPORT_SYMBOL(ieee80211_wx_get_power);
 EXPORT_SYMBOL(ieee80211_wlan_frequencies);
 EXPORT_SYMBOL(ieee80211_wx_set_rts);
 EXPORT_SYMBOL(ieee80211_wx_get_rts);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_tx.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_tx.c
index b29c36b..24035b4 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_tx.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_tx.c
@@ -909,4 +909,6 @@ int ieee80211_xmit(struct sk_buff *skb, struct net_device *dev)
 
 }
 
+/*
 EXPORT_SYMBOL(ieee80211_txb_free);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
index eb085d4..e09a8d9 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
@@ -866,13 +866,19 @@ int ieee80211_wx_set_gen_ie(struct ieee80211_device *ieee, u8 *ie, size_t len)
 
 }
 
+/*
 EXPORT_SYMBOL(ieee80211_wx_set_gen_ie);
+*/
 #if (WIRELESS_EXT >= 18)
+/*
 EXPORT_SYMBOL(ieee80211_wx_set_mlme);
 EXPORT_SYMBOL(ieee80211_wx_set_auth);
 EXPORT_SYMBOL(ieee80211_wx_set_encode_ext);
 EXPORT_SYMBOL(ieee80211_wx_get_encode_ext);
+*/
 #endif
+/*
 EXPORT_SYMBOL(ieee80211_wx_get_scan);
 EXPORT_SYMBOL(ieee80211_wx_set_encode);
 EXPORT_SYMBOL(ieee80211_wx_get_encode);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/rtl819x_HTProc.c b/drivers/staging/rtl8192u/ieee80211/rtl819x_HTProc.c
index 642635f..d7450c4 100644
--- a/drivers/staging/rtl8192u/ieee80211/rtl819x_HTProc.c
+++ b/drivers/staging/rtl8192u/ieee80211/rtl819x_HTProc.c
@@ -1459,4 +1459,6 @@ void HTSetConnectBwModeCallback(struct ieee80211_device* ieee)
 	pHTInfo->bSwBwInProgress = false;
 }
 
+/*
 EXPORT_SYMBOL(HTUpdateSelfAndPeerSetting);
+*/
diff --git a/drivers/staging/rtl8192u/ieee80211/rtl819x_TSProc.c b/drivers/staging/rtl8192u/ieee80211/rtl819x_TSProc.c
index 5373d56..72b6b54 100644
--- a/drivers/staging/rtl8192u/ieee80211/rtl819x_TSProc.c
+++ b/drivers/staging/rtl8192u/ieee80211/rtl819x_TSProc.c
@@ -625,4 +625,6 @@ void TsStartAddBaProcess(struct ieee80211_device* ieee, PTX_TS_RECORD	pTxTS)
 	else
 		IEEE80211_DEBUG(IEEE80211_DL_ERR, "%s()==>BA timer is already added\n", __FUNCTION__);
 }
+/*
 EXPORT_SYMBOL(RemovePeerTS);
+*/
-- 
1.7.1

